chrome/updater/updater_module.cc

```diff
@@ -107,8 +107,15 @@ void Observer::OnEvent(Events event, const std::string& id) {
           updater_status_string_map.find(status_iterator->second)->second);
     }
     if (crx_update_item_.state == ComponentState::kUpdateError) {
-      status +=
-          ", error code is " + std::to_string(crx_update_item_.error_code);
+      // QUICK_ROLL_FORWARD update, adjust the message to "Update installed,
+      // pending restart"
+      if (crx_update_item_.error_code == 21) {
+        status = std::string(
+            updater_status_string_map.find(UpdaterStatus::kUpdated)->second);
```

Comment:
If |updater_status_string_map| is a map, why not just do here `updater_status_string_map[UpdaterStatus::kUpdated]` ?

Reply:
Thanks for the suggestion. Similar to the single-line statement w/ or w/o braces, I saw both styles in our code base:
1. Using `->second`: https://source.corp.google.com/search?q=%22-%3Esecond%22&sq=repo:lbshell-internal%2Fcobalt_src%20AND%20branch:25.lts.1%2B
2. Using `map[`: https://source.corp.google.com/search?q=%22map%5B%22&sq=repo:lbshell-internal%2Fcobalt_src%20AND%20branch:25.lts.1%2B

I don't have a strong preference of one way to another. If you strongly suggest the other way, I can change this reference in this file. But to systematically improve consistency about the usage, we should use Deco Tech Talk to align on the style, clearly document which style is preferred and use follow-up CLs to normalize the usage/format.

Reply:
I personally prefer the second (operator[]) because it makes things look like a map/dictionary, whereas the first is just a find() operator that then returns a pair of which the `second` needs to be taken; this `second` is semantically abstract. TBH I had never seen `find()->second` used in Chromium before, the only cases of using find() I've seen is when both the key and the value are needed.

I hear you're volunteering to give this tech talk or write the discussion document? : )

Reply:
I personally prefer the second (`operator[]`); the `find()->second` makes a map look like is a normal container filled up with pairs (this happens to be often the implementation, but this should be abstracted away). It's true that `operator[]` performs an insertion if the key is not present (which I think it's a huge mistake by STL), in that case `std::map::at()` is still more semantically "meaningful" than `find()`.
FTR the only cases of direct `find()` I remember in Chromium are those when both the key and the value are needed.

OTOH, do I hear you volunteering to present this question to cobalt in a document or (mini) style tech talk? : )

Reply:
Thanks for the explanation. Makes sense, I'll use `operator[]` or `std::map::at()` here. Since the function already checks if the passed in map key exists, it'll be fine to use both.

About code styles, I found this Cobalt page: https://g3doc.corp.google.com/company/teams/cobalt/team/coding_style.md?cl=head. I'll see how to incorporate our discussion here.